File: all/t149/typescript/signature.ts
/**
 * Converts an HSL color value to RGB.
 * Assumes h, s, and l are contained in the set [0, 1] and
 * returns r, g, and b in the set [0, 255].
 *
 * @param hue The hue of the color (0-360)
 * @param saturation The saturation of the color (0-1)
 * @param lightness The lightness of the color (0-1)
 * @return An object containing the red, green, and blue channels.
 */
function hslToRgb(hue: number, saturation: number, lightness: number): { r: number, g: number, b: number } {
}

File: all/t43/c&cpp/signature.cpp
/**
 * Convert RGB color to HSV color.
 * For example:
 *     input: 0, 0, 255
 *     output: 240, 100, 100
 *
 * @param r The red component of the RGB color (integer between 0 and 255)
 * @param g The green component of the RGB color (integer between 0 and 255)
 * @param b The blue component of the RGB color (integer between 0 and 255)
 *
 * @return A tuple containing the HSV values (hue, saturation, value), where hue is in degrees (0-360),
 *         saturation and value are percentages (0-100).
 */
std::tuple<int, int, int> rgb_to_hsv(int r, int g, int b);